ipn/ipnlocal/local_test.go

```diff
@@ -638,3 +639,86 @@ func TestInternalAndExternalInterfaces(t *testing.T) {
 		})
 	}
 }
+
+func TestPacketFilterPermitsUnlockedNodes(t *testing.T) {
+	tests := []struct {
+		name   string
+		peers  []*tailcfg.Node
+		filter []filter.Match
+		want   bool
+	}{
+		{
+			name: "empty",
+			want: false,
+		},
+		{
+			name: "no-unsigned",
+			peers: []*tailcfg.Node{
+				{ID: 1},
+			},
+			want: false,
+		},
+		{
+			name: "unsigned-good",
+			peers: []*tailcfg.Node{
+				{ID: 1, UnsignedPeerAPIOnly: true},
+			},
+			want: false,
+		},
+		{
+			name: "unsigned-bad",
```

Comment:
For completeness, maybe add a 'bad' case where the `filter.Match.Srcs` is a superset of the peer's `AllowedIP` ?

Reply:
done